docs/specification/draft/client/elicitation.mdx

```diff
@@ -321,12 +526,212 @@ Servers should handle each state appropriately:
+- Status of resource access (e.g., API keys or a payment flow via url elicitation)
+
+Servers implementing elicitation **MUST** securely associate this state with individual users following the guidelines in the [security best practices](../basic/security_best_practices) document. Specifically:
+
+- State **MUST NOT** be associated with session IDs alone
+- State storage **MUST** be protected against unauthorized access
+- When using remote communication, user identification **MUST** be derived from authenticated tokens acquired via [MCP authorization](../basic/authorization)
+
+<Note>
+  The examples in this section are non-normative and illustrate potential uses
+  of elicitation. Implementers should adapt these patterns to their specific
+  requirements while maintaining security best practices.
+</Note>
+
+### URL Elicitation for Sensitive Data
+
+For servers that interact with APIs requiring sensitive credentials (e.g. LLM APIs), url elicitation provides a secure mechanism for collecting API keys without exposing them to the MCP client or host application.
+
+In this pattern:
+
+1. The server directs users to a secure web page (served over HTTPS)
+2. The page presents a branded form UI on a domain the user trusts
+3. Users enter sensitive credentials directly into the secure form
+4. The server stores credentials securely, bound to the user's identity
+5. Subsequent MCP requests use these stored credentials for API access
+
+This approach ensures that sensitive credentials never pass through the MCP client or any intermediate MCP servers, reducing the risk of exposure through client-side logging or other attack vectors.
+
+### URL Elicitation for OAuth Flows
+
+URL elicitation enables a pattern where MCP servers act as OAuth clients to third-party resource servers.
+The "downstream authorization" enabled by url elicitation is fundamentally separate from [MCP authorization](../basic/authorization).
+
+<Note>
+  This distinction is critical to understand. URL elicitation is *not* about
+  authorizing (or re-authorizing) the MCP client to access the MCP server. It
+  *can* be used to enable the MCP server to access third-party resources or APIs
+  on behalf of the user, without exposing sensitive credentials or tokens to the
+  MCP client.
+</Note>
+
+#### Understanding the Distinction
+
+- **MCP Authorization**: Required OAuth flow between the MCP client and MCP server (covered in the [authorization specification](../basic/authorization))
+- **Downstream Authorization**: Optional authorization between the MCP server and a third-party resource server, initiated via url elicitation
+
+In downstream authorization, the server acts as both:
+
+- An OAuth resource server (to the MCP client)
+- An OAuth client (to the third-party resource server)
+
+Consider this hypothetical scenario:
+
+- Claude Desktop (the MCP client) connects to Bob's Productivity Server (an HTTPS MCP server)
```

Comment:
We can make this a bit more generic for spec purposes. Something like this:

1. MCP client connects to Example MCP Server
2. Example MCP Server uses downstream payment API that requires separate authorization
3. When MCP client invokes a tool from Example MCP Server that requires payment details, the MCP server needs to request payment API credentials

Reply:
I noticed a lot of confusion when we used generic language and much better understanding with a specific example. But point taken, will make it a bit less specific.